### ISSUE TITLE
Unable to darken an image using a matrix with negative values via feColorMatrix

### ISSUE THREAD

**Issue by tm93112** [canvg/canvg#1111]
Using the following svg as an example, you can see that the image is not rendered darker. 
```
<svg>
  <defs>
         <filter id="darken">
		<feColorMatrix in="SourceGraphic" type="matrix" values="1.14 0 0 -0.7325 0, 0 1.14 0 -0.7325 0, 0 0 1.14 -0.7325 0, 0 0 0 1 0" />
	</filter>
  </defs>
  <rect fill="gray" x="1" y="1" width="100" height="100" filter="url(#darken)" />	
</svg>
```
![darker](https://user-images.githubusercontent.com/19276230/84684775-00f92980-aeee-11ea-9433-b6a87687a69b.PNG)

It appears the ternary in this function is the reason: 
```
function m(
	matrix: number[],
	i: number,
	v: number
) {

	const mi = matrix[i];

	return mi * (mi < 0 ? v - 255 : v);
}
```

If this is changed to 
```
function m(
	matrix: number[],
	i: number,
	v: number
) {

	const mi = matrix[i];

	return mi * v;
}
```
then the darkness is applied appropriately. 

Perhaps there's a reason for this ternary that I'm not aware of from my limited use with canvg that I just can't see?

**Comment by gabelerner** [comment]
This seems valid. We can open a PR around this at some point but also are accepting PRs if somebody has the time to make one here.

**Comment by tm93112** [comment]
Opened https://github.com/canvg/canvg/pull/1117 to address this

**Comment by dangreen** [comment]
Fixed in [v3.0.7](https://github.com/canvg/canvg/releases/tag/v3.0.7)
